<a href="https://colab.research.google.com/github/ghaida-alsalamah/GoRiyadh-Project/blob/main/Task_2_Hotels.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

مراجعة وتحليل بيانات فنادق الرياض (Riyadh Hotels Data) Review

الهدف: فحص جودة البيانات وتحديد المشاكل والنقص قبل البدء بعمليات التحليل المتقدمة.

In [ ]:
import pandas as pd


df = pd.read_excel('Riyadh_Hotels.xlsx')


print("--- عرض عينة من البيانات ---")
display(df.head())

print("\n--- فحص القيم المفقودة ---")
print(df.isnull().sum())


print("\n--- عدد الصفوف المكررة ---")
print(df.duplicated().sum())

--- عرض عينة من البيانات ---


hotel_id                                  hotel_name         SOURCE  \
0  15215188                             Fairmont Riyadh  all.accor.com   
1   9702375                      Rosh Rayhaan by Rotana  Official Site   
2  15756839            Hilton Riyadh Hotel & Residences    Booking.com   
3  24143026                               Joudyan Hotel     MakeMyTrip   
4    969150  Holiday Inn Riyadh - al Qasr, an IHG hotel    Booking.com   

     price  base_price     checkIn    checkOut  count  rating  \
0  1605.00         NaN  2025-01-02  2025-01-03    220     4.5   
1   813.75      693.75  2025-01-02  2025-01-03   1311     4.5   
2  1128.75     1128.75  2025-01-02  2025-01-03   1626     4.5   
3   588.75      472.50  2025-01-02  2025-01-03    317     5.0   
4   633.75      633.75  2025-01-02  2025-01-03   1073     4.5   

                                                Info   latitude  longitude  
0                                                NaN  24.808397  46.718050  
1  ['Free breakfast available', 'Free cancellation']  24.700346  46.682137  
2                                                NaN  24.785522  46.727880  
3  ['Free breakfast available', 'Free cancellation']  24.713552  46.675297  
4                                                NaN  24.683914  46.688030


--- فحص القيم المفقودة ---
hotel_id        0
hotel_name      0
SOURCE          0
price          60
base_price    249
checkIn         0
checkOut        0
count           0
rating          0
Info          758
latitude      438
longitude     438
dtype: int64

--- عدد الصفوف المكررة ---
22


In [ ]:

print(df.dtypes)

hotel_id        int64
hotel_name     object
SOURCE         object
price         float64
base_price    float64
checkIn        object
checkOut       object
count           int64
rating        float64
Info           object
latitude      float64
longitude     float64
dtype: object


هيكلة البيانات: الملف يحتوي على أعمدة أساسية (الاسم، السعر، التقييم

القيم المفقودة (Missing Values):لا يوجد نقص في بيانات التقييمات،جميع بيانات التقييم كامله ١٠٠٪

الصفوف المكررة (Duplicates):تم رصد (٢٢) صفوف مكررة ويجب حذفها لتجنب تكرار النتايج

نوع البيانات (Data Types):عمود السعر سليم تقنياً ومخزن كأرقام عشرية (float64)، مما يسمح لنا بإجراء العمليات الإحصائية مباشرة.

ومع ذلك، يوجد 60 قيمة مفقودة في عمود price و 249 في base_price يجب أخذها في الاعتبار عند حساب المتوسطات.



النقص الكبير (Missing Data):هناك نقص حاد في بيانات الموقع الجغرافي (latitude & longitude) بواقع 438 فندق، ونقص في عمود المعلومات Info بواقع 758 صف. هذا يصعب عملية الخرائط أو تقديم وصف مفصل للفنادق.

In [ ]:

df_cleaned = df.drop_duplicates()


print("عدد الصفوف المكررة بعد التنظيف:", df_cleaned.duplicated().sum())

عدد الصفوف المكررة بعد التنظيف: 0
